--- ../xsconsole-10.1.11-copy/XSConsoleData.py	2021-01-20 16:05:28.397607723 +0100
+++ ./XSConsoleData.py	2021-02-16 15:31:03.761614160 +0100
@@ -522,18 +522,16 @@
         # Double-check authentication
         Auth.Inst().AssertAuthenticated()

-        file = None
-        try:
-            file = open("/etc/resolv.conf", "w")
-            now = datetime.datetime.now().strftime("; created by xsconsole %I:%M%p on %B %d, %Y\n")
-            file.write(now)
-            for other in self.dns.othercontents([]):
-                file.write(other+"\n")
-            for server in self.dns.nameservers([]):
-                file.write("nameserver "+server+"\n")
-        finally:
-            if file is not None: file.close()
-            self.UpdateFromResolveConf()
+        for pif in Data.Inst().derived.managementpifs([]):
+            ipv6 = pif['primary_address_type'].lower() == 'ipv6'
+            mode = pif['ipv6_configuration_mode'] if ipv6 else pif['ip_configuration_mode']
+            ip = pif['IPv6'][0].split('/')[0] if ipv6 else pif['IP']
+            netmask = pif['IPv6'][0].split('/')[1] if ipv6 else pif['netmask']
+            gw = pif['ipv6_gateway'] if ipv6 else pif['gateway']
+            dns = ','.join(self.dns.nameservers([]))
+            self.ReconfigureManagement(pif, mode, ip, netmask, gw, dns)
+
+        self.UpdateFromResolveConf()
